Build fails with error "opencv2/gapi.hpp: No such file or directory"

When I build opencv 4.x on my ubuntu20.04, I get the following Error:
fatal error: opencv2/gapi.hpp: No such file or directory
    1 | #include <opencv4/gapi.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/build.make:63: samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:20543: samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

Could anyone help to solve this problem? thanks~

Hitting this with both 4.5.5 and 4.6.0 when building *with* examples, so the obvious workaround is to disable examples:

For 4.6.0:
[1680/1854] /usr/bin/x86_64-pc-linux-gnu-g++ -DHAVE_THREADS=1 -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/photo/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/imgproc/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/core/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/objdetect/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/calib3d/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/imgcodecs/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/features2d/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/dnn/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/flann/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/video/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/videoio/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/highgui/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/datasets/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/text/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/plot/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/ml/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/optflow/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/ximgproc/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/phase_unwrapping/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/tracking/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/quality/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/xphoto/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/xobjdetect/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/wechat_qrcode/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/videostab/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/surface_matching/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/superres/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/structured_light/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/stereo/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/shape/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/saliency/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/rgbd/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/reg/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/rapid/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/mcc/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/line_descriptor/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/intensity_transform/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/img_hash/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/hfs/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/fuzzy/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/face/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/dpm/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/dnn_superres/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/dnn_objdetect/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/ccalib/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/bioinspired/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/bgsegm/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/barcode/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/aruco/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/alphamat/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/stitching/include -isystem /var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0_build-abi_x86_64.amd64 -isystem /usr/include/eigen3  -O2 -march=znver3 -pipe -fomit-frame-pointer -g -ggdb3   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mf16c -mfma -mavx -mavx2 -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp -fPIE -std=c++11 -MD -MT samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o -MF samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o.d -o samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o -c /var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/samples/cpp/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp
FAILED: samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DHAVE_THREADS=1 -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/photo/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/imgproc/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/core/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/objdetect/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/calib3d/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/imgcodecs/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/features2d/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/dnn/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/flann/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/video/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/videoio/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/highgui/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/datasets/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/text/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/plot/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/ml/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/optflow/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/ximgproc/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/phase_unwrapping/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/tracking/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/quality/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/xphoto/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/xobjdetect/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/wechat_qrcode/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/videostab/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/surface_matching/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/superres/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/structured_light/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/stereo/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/shape/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/saliency/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/rgbd/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/reg/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/rapid/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/mcc/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/line_descriptor/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/intensity_transform/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/img_hash/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/hfs/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/fuzzy/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/face/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/dpm/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/dnn_superres/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/dnn_objdetect/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/ccalib/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/bioinspired/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/bgsegm/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/barcode/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/aruco/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv_contrib-4.6.0/modules/alphamat/include -I/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/modules/stitching/include -isystem /var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0_build-abi_x86_64.amd64 -isystem /usr/include/eigen3  -O2 -march=znver3 -pipe -fomit-frame-pointer -g -ggdb3   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mf16c -mfma -mavx -mavx2 -fvisibility=hidden -fvisibility-inlines-hidden -fopenmp -fPIE -std=c++11 -MD -MT samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o -MF samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o.d -o samples/cpp/CMakeFiles/example_tutorial_dynamic_graph_snippets.dir/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp.o -c /var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/samples/cpp/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp
/var/tmp/portage/media-libs/opencv-4.6.0-r1/work/opencv-4.6.0/samples/cpp/tutorial_code/gapi/doc_snippets/dynamic_graph_snippets.cpp:1:10: fatal error: opencv2/gapi.hpp: No such file or directory
    1 | #include <opencv2/gapi.hpp>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

See also: https://bugs.gentoo.org/830163

i am having trouble with `cmake` even after adding `opencv_gapi` in file `/opencv4.6.0/samples/cpp/CMakelists.txt`, and `sudo apt-get install libopencv-core-dev`
i am doing it under Python virtual environment
OS: Ubuntu22.04

`cmake -D CMAKE_BUILD_TYPE=RELEASE \
	-D CMAKE_INSTALL_PREFIX=/usr/local \
	-D INSTALL_PYTHON_EXAMPLES=ON \
	-D INSTALL_C_EXAMPLES=OFF \
	-D OPENCV_ENABLE_NONFREE=ON \
	-D WITH_CUDA=ON \
	-D WITH_CUDNN=ON \
	-D OPENCV_DNN_CUDA=ON \
	-D ENABLE_FAST_MATH=1 \
	-D CUDA_FAST_MATH=1 \
	-D CUDA_ARCH_BIN=7.5 \
	-D WITH_CUBLAS=1 \
	-D OPENCV_EXTRA_MODULES_PATH=~/opencv_contrib/modules \
	-D HAVE_opencv_python3=ON \
	-D BUILD_EXAMPLES=ON ..`

end result
`-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
/home/user/opencv/modules/gapi/VA_INCLUDE_DIR
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi
   used as include directory in directory /home/user/opencv/modules/gapi

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.`

No can do:
```
[ 99%] Building CXX object modules/python3/CMakeFiles/opencv_python3.dir/__/src2/cv2.cpp.o
In file included from /home/chowkidar/projects/opencv_quickstart/opencv/modules/python/src2/cv2.cpp:11:
/home/chowkidar/projects/opencv_quickstart/build/modules/python_bindings_generator/pyopencv_generated_include.h:53:10: fatal error: opencv2/gapi/core.hpp: No such file or directory
   53 | #include "opencv2/gapi/core.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [modules/python3/CMakeFiles/opencv_python3.dir/build.make:82: modules/python3/CMakeFiles/opencv_python3.dir/__/src2/cv2.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:8930: modules/python3/CMakeFiles/opencv_python3.dir/all] Error 2
gmake: *** [Makefile:182: all] Error 2
$ uname -a
Linux raspbari14 5.15.84-v7+ #1613 SMP Thu Jan 5 11:59:48 GMT 2023 armv7l GNU/Linux
OS: Raspbian GNU/Linux 11 (bullseye)
$ git status
On branch 4.x
Your branch is up to date with 'origin/4.x'.
```

# Help Target
help:
	@echo "The following are some of the valid targets for this Makefile:"
	@echo "... all (the default if no target is provided)"
	@echo "... clean"
	@echo "... depend"
	@echo "... edit_cache"
	@echo "... install"
	@echo "... install/local"
	@echo "... install/strip"
	@echo "... list_install_components"
	@echo "... package"
	@echo "... package_source"
	@echo "... rebuild_cache"
	@echo "... test"
	@echo "... check_flake8"
	@echo "... check_pylint"
	@echo "... gen_opencv_objc_source"
	@echo "... gen_opencv_objc_source_ios"
	@echo "... gen_opencv_objc_source_osx"
	@echo "... gen_opencv_python_source"
	@echo "... opencv_dnn_plugins"
	@echo "... opencv_highgui_plugins"
	@echo "... opencv_modules"
	@echo "... opencv_perf_tests"
	@echo "... opencv_tests"
	@echo "... opencv_videoio_plugins"
	@echo "... uninstall"
	@echo "... ittnotify"
	@echo "... jsimd"
	@echo "... libjpeg-turbo"
	@echo "... libopenjp2"
	@echo "... libpng"
	@echo "... libprotobuf"
	@echo "... libtiff"
	@echo "... libwebp"
	@echo "... opencv_alphamat"
	@echo "... opencv_annotation"
	@echo "... opencv_aruco"
	@echo "... opencv_barcode"
	@echo "... opencv_bgsegm"
	@echo "... opencv_bioinspired"
	@echo "... opencv_calib3d"
	@echo "... opencv_ccalib"
	@echo "... opencv_core"
	@echo "... opencv_datasets"
	@echo "... opencv_dnn"
	@echo "... opencv_dnn_objdetect"
	@echo "... opencv_dnn_superres"
	@echo "... opencv_dpm"
	@echo "... opencv_face"
	@echo "... opencv_features2d"
	@echo "... opencv_flann"
	@echo "... opencv_freetype"
	@echo "... opencv_fuzzy"
	@echo "... opencv_hdf"
	@echo "... opencv_hfs"
	@echo "... opencv_highgui"
	@echo "... opencv_img_hash"
	@echo "... opencv_imgcodecs"
	@echo "... opencv_imgproc"
	@echo "... opencv_intensity_transform"
	@echo "... opencv_interactive-calibration"
	@echo "... opencv_line_descriptor"
	@echo "... opencv_mcc"
	@echo "... opencv_ml"
	@echo "... opencv_model_diagnostics"
	@echo "... opencv_objdetect"
	@echo "... opencv_optflow"
	@echo "... opencv_perf_aruco"
	@echo "... opencv_perf_bioinspired"
	@echo "... opencv_perf_calib3d"
	@echo "... opencv_perf_core"
	@echo "... opencv_perf_dnn"
	@echo "... opencv_perf_dnn_superres"
	@echo "... opencv_perf_features2d"
	@echo "... opencv_perf_imgcodecs"
	@echo "... opencv_perf_imgproc"
	@echo "... opencv_perf_line_descriptor"
	@echo "... opencv_perf_objdetect"
	@echo "... opencv_perf_optflow"
	@echo "... opencv_perf_photo"
	@echo "... opencv_perf_reg"
	@echo "... opencv_perf_rgbd"
	@echo "... opencv_perf_stereo"
	@echo "... opencv_perf_stitching"
	@echo "... opencv_perf_superres"
	@echo "... opencv_perf_tracking"
	@echo "... opencv_perf_video"
	@echo "... opencv_perf_videoio"
	@echo "... opencv_perf_xfeatures2d"
	@echo "... opencv_perf_ximgproc"
	@echo "... opencv_perf_xphoto"
	@echo "... opencv_phase_unwrapping"
	@echo "... opencv_photo"
	@echo "... opencv_plot"
	@echo "... opencv_python3"
	@echo "... opencv_quality"
	@echo "... opencv_rapid"
	@echo "... opencv_reg"
	@echo "... opencv_rgbd"
	@echo "... opencv_saliency"
	@echo "... opencv_shape"
	@echo "... opencv_stereo"
	@echo "... opencv_stitching"
	@echo "... opencv_structured_light"
	@echo "... opencv_superres"
	@echo "... opencv_surface_matching"
	@echo "... opencv_test_aruco"
	@echo "... opencv_test_barcode"
	@echo "... opencv_test_bgsegm"
	@echo "... opencv_test_bioinspired"
	@echo "... opencv_test_calib3d"
	@echo "... opencv_test_core"
	@echo "... opencv_test_dnn"
	@echo "... opencv_test_dnn_superres"
	@echo "... opencv_test_face"
	@echo "... opencv_test_features2d"
	@echo "... opencv_test_flann"
	@echo "... opencv_test_freetype"
	@echo "... opencv_test_fuzzy"
	@echo "... opencv_test_hdf"
	@echo "... opencv_test_highgui"
	@echo "... opencv_test_img_hash"
	@echo "... opencv_test_imgcodecs"
	@echo "... opencv_test_imgproc"
	@echo "... opencv_test_intensity_transform"
	@echo "... opencv_test_line_descriptor"
	@echo "... opencv_test_mcc"
	@echo "... opencv_test_ml"
	@echo "... opencv_test_objdetect"
	@echo "... opencv_test_optflow"
	@echo "... opencv_test_phase_unwrapping"
	@echo "... opencv_test_photo"
	@echo "... opencv_test_quality"
	@echo "... opencv_test_rapid"
	@echo "... opencv_test_reg"
	@echo "... opencv_test_rgbd"
	@echo "... opencv_test_saliency"
	@echo "... opencv_test_shape"
	@echo "... opencv_test_stereo"
	@echo "... opencv_test_stitching"
	@echo "... opencv_test_structured_light"
	@echo "... opencv_test_superres"
	@echo "... opencv_test_text"
	@echo "... opencv_test_tracking"
	@echo "... opencv_test_video"
	@echo "... opencv_test_videoio"
	@echo "... opencv_test_videostab"
	@echo "... opencv_test_viz"
	@echo "... opencv_test_wechat_qrcode"
	@echo "... opencv_test_xfeatures2d"
	@echo "... opencv_test_ximgproc"
	@echo "... opencv_test_xphoto"
	@echo "... opencv_text"
	@echo "... opencv_tracking"
	@echo "... opencv_ts"
	@echo "... opencv_version"
	@echo "... opencv_video"
	@echo "... opencv_videoio"
	@echo "... opencv_videostab"
	@echo "... opencv_visualisation"
	@echo "... opencv_viz"
	@echo "... opencv_waldboost_detector"
	@echo "... opencv_wechat_qrcode"
	@echo "... opencv_xfeatures2d"
	@echo "... opencv_ximgproc"
	@echo "... opencv_xobjdetect"
	@echo "... opencv_xphoto"
	@echo "... quirc"
.PHONY : help

--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -15,6 +15,7 @@ set(OPENCV_CPP_SAMPLES_REQUIRED_DEPS
   opencv_calib3d
   opencv_stitching
   opencv_dnn
+  opencv_gapi
   ${OPENCV_MODULES_PUBLIC}
   ${OpenCV_LIB_COMPONENTS})
 ocv_check_dependencies(${OPENCV_CPP_SAMPLES_REQUIRED_DEPS})

